pkgs/by-name/sw/swaylock-plugin/package.nix

```diff
@@ -0,0 +1,87 @@
+{
+  cairo,
+  fetchFromGitHub,
+  gdk-pixbuf,
+  lib,
+  libxcrypt,
+  libxkbcommon,
+  makeWrapper,
+  meson,
+  ninja,
+  nix-update-script,
+  pam,
+  pkg-config,
+  scdoc,
+  stdenv,
+  swaybg,
+  systemd,
+  versionCheckHook,
+  wayland,
+  wayland-protocols,
+  wayland-scanner,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "swaylock-plugin";
+  version = "unstable-2025-01-28";
+  src = fetchFromGitHub {
+    owner = "mstoeckl";
+    repo = "swaylock-plugin";
+    rev = "ac02c528bce8f529f33c85065d77eac1aceccbe5";
+    hash = "sha256-e4iQ7yCPkkJBpgD0aE94lsID8v3kXhr7YmAszfFi7zA=";
+  };
+
+  strictDeps = true;
+  depsBuildBuild = [ pkg-config ];
+  nativeInstallCheckInputs = [
+    versionCheckHook
+  ];
+  nativeBuildInputs = [
+    makeWrapper
+    meson
+    ninja
+    pkg-config
+    scdoc
+    wayland-scanner
+  ];
+  buildInputs = [
+    cairo
+    libxcrypt
+    gdk-pixbuf
+    libxkbcommon
+    pam
+    systemd
+    wayland
+    wayland-protocols
+  ];
+
+  postInstall = ''
+    wrapProgram $out/bin/swaylock-plugin --suffix PATH : ${lib.getBin swaybg}
```

Comment:
Currently seeing

```
$ ./result/bin/swaylock-plugin 
Forked background plugin (pid = 10994): swaybg -c '#ffffff' -o '*' -i '/nix/store/lknmh8fb86r2if9czmr9k57sd8p77syl-pastel.png' -m fill
sh: line 1: swaybg: command not found
```

```suggestion
    wrapProgram $out/bin/swaylock-plugin --suffix PATH : ${lib.getBin swaybg}/bin
```

Reply:
Ah yeah, good point, I should have used makeBinPath here!